specs/DataSet.rst

```diff
@@ -17,6 +17,13 @@ As long as a DataSet is used for data storage, users can freely select the QCoDe
 Terminology
 ================
 
+Metadata
+    Many items in this spec have metadata associated with them.
+    In all cases, we expect metadata to be represented as a dictionary with string keys.
+    While the values are arbitrary and up to the user, in many cases we expect metadata to be nested, string-keyed dictionaries
+    with scalars (strings or numbers) as the final values.
```

Comment:
I would argue that the final values need to support anything that a parameter can return. 

The most direct example is some parameters that are arrays (like a vector of integration weights). We want to store this in the metadata (and are currently able to do so). I think it is important to note this here.